dartagnan/src/main/java/com/dat3m/dartagnan/program/specification/AssertInline.java

```diff
@@ -20,8 +21,10 @@ public AssertInline(Assert assertion){
     @Override
     public BooleanFormula encode(EncodingContext ctx) {
         final BooleanFormulaManager bmgr = ctx.getBooleanFormulaManager();
-        return bmgr.implication(ctx.execution(assertion),
-                ctx.encodeExpressionAsBooleanAt(assertion.getExpression(), assertion));
+        return (ctx.encodeUserAssertions() || assertion.hasTag(Tag.OVERFLOW) || assertion.hasTag(Tag.NULLDEREF))
+                ? bmgr.implication(ctx.execution(assertion),
+                        ctx.encodeExpressionAsBooleanAt(assertion.getExpression(), assertion))
+                : bmgr.makeTrue();
```

Comment:
I think just not encoding the assertions is not right, because we still terminate early and just not report an error. Indeed, I think we could erroneously return PASS in edge-cases where the error we do not want to check for is always triggered.

Rather than changing the encoding, I think it might be better to have a pass that deletes the assertions + the abort, if they do not match with what the user wants to check for. Alternatively, the `Inlining` pass could just never generate the assertions in the first place.
This might be even better because it makes sure that no other pass ever sees the aborts and optimizes around them.

That being said, for user-placed assertions, LLVM generates an `unreachable` instruction which still causes an abort, even if the assertion itself gets deleted before inlining.
On the other hand, we should probably not use the standard assertions provided by `assert.h` and instead define `#define assert(cond, ...) __VERIFIER_assert(cond)` with a custom assert function.